### PR TITLE
Use gRPC ClientConn for otlpmetricgrpc client connection handling

### PR DESCRIPTION
**Pull request by MrAlias**
Follow up to https://github.com/open-telemetry/opentelemetry-go/pull/2329.

This updates the `otlpmetric` module:

- Add common and reusable `retry` internal package.
- Update the `otlpmetricgrpc` to use the gRPC `ClientConn` similar to `otlptracegrpc`.
- Use the new `retry` package in the `otlpmetrichttp` client.

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: exporters/otlp/otlpmetric/otlpmetricgrpc/client.go [hunk 1]
```diff
@@ -105,3 +326,4 @@ func (c *client) UploadMetrics(ctx context.Context, protoMetrics []*metricpb.Res
 	}
 	return err
 }
```
Comment (XSAM) [inline exporters/otlp/otlpmetric/otlpmetricgrpc/client.go:328]: Should we remove these commented codes?

**Review by XSAM — APPROVED** [review]
Looks good overall.

I might miss some contexts here, the implementation of `go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc/client.go` looks similar to `exporters/otlp/otlptrace/otlptracegrpc/client.go`. Any plan to merge these clients? so we do not change twice in the feature.

File: exporters/otlp/otlpmetric/otlpmetricgrpc/client.go (hunk 1 above)
Comment (MrAlias) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/client.go]: Gah, yes indeed. Thanks for catching this.

**Comment by MrAlias** [comment]
> I might miss some contexts here, the implementation of `go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc/client.go` looks similar to `exporters/otlp/otlptrace/otlptracegrpc/client.go`. Any plan to merge these clients? so we do not change twice in the feature.

No, you didn't miss something and you are not alone (https://github.com/open-telemetry/opentelemetry-go/issues/2015). Ever since we refactored the OTLP exporters to be split by signal this duplication has existed. I think the next step now that this will bring the two signals in line is to unify and obscure common code in `internal` packages/modules. We will likely need to obscure things in internal modules so the whole project does not depend on gRPC or the backoff packages.

**Pull request merged**